pkg/stub/imagestreams.go

```diff
@@ -221,41 +221,44 @@ func (h *Handler) upsertImageStream(imagestreamInOperatorImage, imagestreamInClu
 }
 
 func (h *Handler) updateDockerPullSpec(oldies []string, imagestream *imagev1.ImageStream, opcfg *v1.Config) {
-	//TODO remove when TBR creds sorted out and we do not explicitly set SamplesRegistry to registry.access.redhat.com
-	// by default we want to leave the jenkins images as using the payload set to the IMAGE* env's
-	// but if the customer overrides (after we switch to TBR and don't set SamplesRegistry by default), we'll let them
-	// change jenkins related images
-	switch imagestream.Name {
-	case "jenkins":
-		return
-	case "jenkins-agent-nodejs":
-		return
-	case "jenkins-agent-maven":
-		return
-	}
-	if len(opcfg.Spec.SamplesRegistry) > 0 {
-		logrus.Debugf("updateDockerPullSpec stream %s has repo %s", imagestream.Name, imagestream.Spec.DockerImageRepository)
-		// don't mess with deprecated field unless it is actually set with something
-		if len(imagestream.Spec.DockerImageRepository) > 0 &&
-			!strings.HasPrefix(imagestream.Spec.DockerImageRepository, opcfg.Spec.SamplesRegistry) {
-			// if not one of our 4 defaults ...
-			imagestream.Spec.DockerImageRepository = h.coreUpdateDockerPullSpec(imagestream.Spec.DockerImageRepository,
-				opcfg.Spec.SamplesRegistry,
-				oldies)
+	registryReplacement := "registry.redhat.io"
```

Comment:
isn't this going to default into replacing everything with registry.redhat.io even if they specified no replacement?  why do we want that?

Reply:
this is how it defaults to registry.redhat.io

Reply:
it will now look like:

```
	switch imagestream.Name {
	case "jenkins":
		return
	case "jenkins-agent-nodejs":
		return
	case "jenkins-agent-maven":
		return
	}

	registryReplacement := "registry.redhat.io"
	if len(opcfg.Spec.SamplesRegistry) != 0 {
		registryReplacement = opcfg.Spec.SamplesRegistry
	}
```

As a reminder, back in the early days, we did this as a way to catch any stray imagestreams still pointing to docker.io or registry.access.redhat.com, etc.

Reply:
I don't think we want it to default to anything.  If the user didn't provide an alternate registry value, we should leave the imagestreams alone.  If they did provide one, we should use the value they provided to replace our canonical list, except for jenkins.

> As a reminder, back in the early days, we did this as a way to catch any stray imagestreams still pointing to docker.io or registry.access.redhat.com, etc.

hm, I don't recall that and i'm not sure why we we'd want to change them?  If they're pointing there, we should fix the imagestream content.  Otherwise, if it points to docker.io there may well be a good reason (it may not be on registry.redhat.io).

Reply:
the user doesn't provide these ... this comes from the openshift/library content we pull in

unless the user overrides the registry, he expects it to be correct

Reply:
> the user doesn't provide these ... this comes from the openshift/library content we pull in

right, hence my question above of why we'd want to change them, absent an explicit user-provided alternate registry to use?

> unless the user overrides the registry, he expects it to be correct

Exactly, and the samples operator should also expect them to be correct.  We don't own the imagestreams, we get them from elsewhere, so if someone provides us with an imagestream and says "this comes from docker.io" (or quay.io, or something else) we should be respecting that value, absent an explicit override from the user.  Defaulting the override registry hostname to "registry.redhat.io" (or any other value) doesn't make sense to me...

Reply:
OK, yeah there was originally a concern with openshift/library OCP content providing non TBR content and we should catch that.  And the code was doing that previously.

I'm fine with moving off of that now as part of this PR.  I'll push a change in a bit.

Reply:
> OK, yeah there was originally a concern with openshift/library OCP content providing non TBR content and we should catch that. And the code was doing that previously.

yeah i can't remember that convo or the context, but let's go with "i've come to a new understanding and changed my mind" :)